kuka_rox_hardware_interface/src/robot_manager_node.cpp

```diff
@@ -144,15 +171,22 @@ RobotManagerNode::on_activate(const rclcpp_lifecycle::State &)
     return FAILURE;
   }
 
-  // auto position_controller_name = this->get_parameter("position_controller_name").as_string();
-  // auto impedance_controller_name = this->get_parameter("impedance_controller_name").as_string();
-  // controller_name_ = (this->get_parameter("command_mode").as_string() == "position")
-  //     ? position_controller_name : torque_controller_name;
-  controller_names_.clear();
-  controller_names_.emplace_back("joint_trajectory_controller");
-  if (is_joint_imp_contr_) {
-    controller_names_.emplace_back("joint_impedance_controller");
+  // Select control modes
+  auto position_controller_name = this->get_parameter(POSITION_CONTROLLER_NAME_PARAM).as_string();
+  auto impedance_controller_name = this->get_parameter(IMPEDANCE_CONTROLLER_NAME_PARAM).as_string();
```

Comment:
With this map structure I think it would be a little better to fill the map when the name changes occur, that way you do not have to fill all fields of the map unnecessarily.

I know we agreed on this way previously, but if we assume that no one changes the controller names just for fun, filling the map only in the name changes is a little bit more efficient in my opinion.

In the future we may also want to allow control mode changes if external control is active, then this solution has too much overhead.
What are your thoughts @altex111 @VX792 @kovacsge11 ?

Reply:
This function is only called once so performance-wise it might not be a problem, but I think it would make more sense to do this in the on_configure() function. For one,  reading parameters and creating a data structure which stores a part of the predefined yaml has more to do with configuration than activation, and we also aren't delaying the first message by doing stuff like dynamically allocating vectors and filling the map.

But I think you're right, the solution depends on how changing the control mode in active state will be implemented

Reply:
[Result]: control_mode_map refill in callbacks

Reply:
I think you do not need the names and map filling here, only the map's find line